source/common/http/http2/codec_impl.cc

```diff
@@ -1431,6 +1433,52 @@ void ConnectionImpl::dumpState(std::ostream& os, int indent_level) const {
   }
 }
 
+void ConnectionImpl::dumpStreams(std::ostream& os, int indent_level) const {
+  const char* spaces = spacesForLevel(indent_level);
+
+  // Try to dump details for the current stream.
+  // If none, dump a subset of our active streams.
+  os << spaces << "Number of active streams: " << active_streams_.size()
+     << DUMP_OPTIONAL_MEMBER(current_stream_id_);
+
+  if (current_stream_id_.has_value()) {
+    os << " Dumping current stream:\n";
+    const ConnectionImpl::StreamImpl* stream = getConstStream(current_stream_id_.value());
+    DUMP_DETAILS(stream);
+  } else {
+    os << " Dumping " << std::min<size_t>(25, active_streams_.size()) << " Active Streams:\n";
```

Comment:
Any specific reason to drop this down to 25? Mostly curious, I don't have a strong opinion on the right value

Reply:
Seemed like it could get rather spammy. If it becomes an issue, we could make it configurable.